File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s=s.toLowerCase();
        int left=0;
        int right=s.length()-1;
        
        while(left<right){
            while(left<right && !Character.isLetterOrDigit(s.charAt(left))){left++;}
            while(left<right && !Character.isLetterOrDigit(s.charAt(right))){ right--; }
            
            if(left< right && s.charAt(left)!=s.charAt(right)) { return false; }
            left++;
            right--;
        }
        return true;
    }
}

/*
Input: s = "A man, a plan, a canal: Panama"
Output: true
Explanation: "amanaplanacanalpanama" is a palindrome.

Input: s = "race a car"
Output: false
Explanation: "raceacar" is not a palindrome.

*/

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        for(int i=0; i<s.size();i++)
        {
            if(s[i]>='A' && s[i]<='Z')
            {
                char temp=s[i]+32;
                s[i]=temp;
            }
        }
        string result="";
        for(int i=0; i<s.size();i++)
        {
            if((s[i]>='a' &&s[i]<='z' || (s[i]>='0' && s[i]<='9')))
            {
                result.push_back(s[i]);
            }
        }
        
        int i=0, j=result.size()-1;
        while(i<j)
        {
            if(result[i]==result[j])
            {
                i++;
                j--;
            }
            else
            {
                return false;
            }
        }
        return true;
    }
};